frontend/src/util/shapes.ts

```diff
@@ -0,0 +1,23 @@
+import cytoscape from "cytoscape";
+
+export type NodeShape = {
+  name: string;
+  points?: number[];
```

Comment:
Are you sure these type defs can't be extracted from Cytoscape directly, instead of redefining them?

Reply:
After checking the cytoscape type definition, I didn't see any types matching. The type definition returned when cytoscape is initialized with two string arguements is `unknown`. Also, in the cytoscape's type defintion file, there exists a `nodeShapes` type but it doesn't match the returned type from:
```ts
const shapes = baseRenderer.prototype.nodeShapes;
```
But please I'm fully open to suggestions.Thank you.

Reply:
Thank you for the question. 

The current type definitions we're using are partially from Cytoscape and partially custom because:

1. For the shape names, we're using Cytoscape's built-in type `cytoscape.Css.NodeShape`

2. We need to keep `CytoscapeRenderer` as a custom type because:
   - When `cytoscape("renderer", "base")` is called, it returns `unknown`
   - We need to type this return value to access `prototype.nodeShapes` safely
   - This internal renderer API isn't exposed in Cytoscape's public type definitions
   - Since we're accessing internal implementation details to extract the shapes, we need to define these types ourselves

3. The `points` field type (`number[]`) comes from the actual implementation rather than Cytoscape's type definitions, which suggest it should be a string via `cytoscape.Css.ShapePolygonPoints`. 

(I think keeping the internal implementation of `number[]` aligns with the actual usage and makes more sense than strictly following their type definition of space-separated string in this case. But I'm fully open to suggestions/feedback/concerns on the decision).

Reply:
Thanks for leaving a detailed reply. This all makes sense. I ended up taking a look at this myself, and yeah there didn't seem to be a good way to extract the types from the library.

Take a look at what I ended up doing in #41. After trying to make use of the `points` field directly from cytoscape, I decided that it was a little too messy, and the shapes didn't even look that good anyway. It turns out not all of the `points` were `number[]`, some were functions to generate points, and some were undefined.

So instead, I just made use of cytoscapes custom polygon shape feature and defined my own shapes that I liked better anyway.